Added alerts pattern.

dsjen opened the pull request:
## [UITK-47](https://openedx.atlassian.net/browse/UITK-47)

Added alerts pattern.

![screen shot 2016-01-28 at 2 03 41 pm](https://cloud.githubusercontent.com/assets/5265058/12655944/7d08dace-c5c9-11e5-8f49-5f0eb713b143.png)
![screen shot 2016-01-28 at 2 03 12 pm](https://cloud.githubusercontent.com/assets/5265058/12655946/7d0be2dc-c5c9-11e5-85c2-7568b3471f36.png)
![screen shot 2016-01-28 at 2 03 48 pm](https://cloud.githubusercontent.com/assets/5265058/12655945/7d0b25c2-c5c9-11e5-9589-ffaa6f0ea245.png)

## Still TODO:
- [x] The post will likely need to be pared down a bit as it includes three examples per alert style.
- [x] Titles on examples need to be updated.
- [x] @chris-mike -- description of alerts pattern needs to be updated.

### Testing
- [x] RTL

### Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [x] Code review: @andy-armstrong 
- ~~UI/Visual Design & Code review: @talbs~~ N/A
- [x] Accessibility review: @clytwynec @clrux

### Post-review
- [x] Squash commits
- [ ] Publish packages/doc site

dsjen commented:
FYI, the `.css` files are auto-generated by SASS.

talbs commented:
FYI, @chris-mike. Here's the PR for the Alerts pattern we've been working on. No need to review the code, but we should review the rendered UI with @dsjen in person. :)

dsjen commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,275 @@
+
+description:    Examples of alerts in a semi-real world context.
+
+info:           Alerts provide information about...TBD
```
@chris-mike -- this text is what is displayed in the PL documentation site.  Please let me know what this can be updated to.  E.g. [UXPL - Button](http://ux.edx.org/#buttons).

talbs commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,275 @@
+<h3 class="hd-6 example-set-hd">Error w/ Title</h3>
+<div class="example-set">
+    <div class="alert alert-error">
+        <div class="icon-display">
```
Hmm, looks like you're following/copying from our example icon markup here with the ``<div class="icon-display">``. This element isn't needed to display icons and can be removed. I'd rely on just styling the ``.alert-icon`` element you have nested in here.

dsjen replied:
Thanks, updated.

talbs commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,275 @@
+                Alert Title!
+            </div>
+
+            <div class="alert-copy">
```
I think we can use a more semantic tag for this example. Assuming an alert's message would/should never go beyond a paragraph's length, we can use a ``<p>`` here.

dsjen replied:
Makes sense -- updated.

talbs commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,275 @@
+        </div>
+
+        <div class="alert-message">
+            <div class="alert-title">
```
We should use a more semantic element here. I'd recommend a heading (anything from ``<h1>`` to ``<h6>``, as in an app's UI, a dev/designer may need to use any or all of them based on their view's semantic outline/tree).

dsjen replied:
Updated.

talbs commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,275 @@
+            </div>
+
+            <div class="alert-copy">
+                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut
```
NIT: We should try to populate this with content from one of the real alert cases we identified. It could help suggest length, tone, and general purpose of this element.

dsjen replied:
Updated.

talbs commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,275 @@
+
+        <div class="alert-message">
+            <div class="alert-title">
+                Alert Title!
```
NIT: We should try to populate this with content from one of the real alert cases we identified. It could help suggest length, tone, and general purpose of this element.

dsjen replied:
Updated.

talbs commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,275 @@
+        </div>
+
+        <div class="alert-action">
+            <button type="button" class="btn-neutral btn">Alert Action</button>
```
NIT: We should try to populate this with content from one of the real alert cases we identified. It could help suggest length, tone, and general purpose of this element.

dsjen replied:
Updated.

talbs commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,275 @@
+            </div>
+        </div>
+
+        <div class="alert-action">
```
Is this ``<div class="alert-action">`` needed for layout purposes or could you get away with just styling the ``<button class="alert-action">`` instead?

dsjen replied:
Good point!  Updated.

talbs commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,275 @@
+
+
+<h3 class="hd-6 example-set-hd">Error w/o Title</h3>
+<div class="example-set">
```
These examples could benefit from the ``<div class="example-set-description">`` element we've added into the UXPL Doc Site in case examples need a description. Mind moving the "Error w/o Title" and other similarly minded context into a ``<p>`` here?

dsjen replied:
I'm not sure I totally follow.  Are you saying that the header in this case should be a description?  I looked at http://ux.edx.org/#headings as an example of the header and description usage, but it's not entirely clear to me what the description in the case of these examples would be.

talbs commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,275 @@
+<h3 class="hd-6 example-set-hd">Error w/o Title</h3>
+<div class="example-set">
+    <div class="alert alert-error">
+        <div class="icon-display">
```
See [my note about simplifying this above](https://github.com/edx/ux-pattern-library/pull/277/files#r51175181).

dsjen replied:
Thanks, done.

talbs commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,275 @@
+<h3 class="hd-6 example-set-hd">Error w/o Title</h3>
+<div class="example-set">
+    <div class="alert alert-error">
+        <div class="icon-display">
```
See [my note about simplifying this above](https://github.com/edx/ux-pattern-library/pull/277/files#r51175181).

dsjen replied:
Done.

talbs commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,275 @@
+<h3 class="hd-6 example-set-hd">Success w/ Title</h3>
+<div class="example-set">
+    <div class="alert alert-success">
+        <div class="icon-display">
```
See [my note about simplifying this above](https://github.com/edx/ux-pattern-library/pull/277/files#r51175181).

dsjen replied:
Done.

talbs commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,275 @@
+<h3 class="hd-6 example-set-hd">Success w/o Title</h3>
+<div class="example-set">
+    <div class="alert alert-success">
+        <div class="icon-display">
```
See [my note about simplifying this above](https://github.com/edx/ux-pattern-library/pull/277/files#r51175181).

dsjen replied:
Done.

talbs commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,275 @@
+<h3 class="hd-6 example-set-hd">Success w/o Title</h3>
+<div class="example-set">
+    <div class="alert alert-success">
+        <div class="icon-display">
```
See [my note about simplifying this above](https://github.com/edx/ux-pattern-library/pull/277/files#r51175181).

dsjen replied:
Done.

talbs commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,275 @@
+<h3 class="hd-6 example-set-hd">Warning w/ Title</h3>
+<div class="example-set">
+    <div class="alert alert-warning">
+        <div class="icon-display">
```
See [my note about simplifying this above](https://github.com/edx/ux-pattern-library/pull/277/files#r51175181).

dsjen replied:
Done.

talbs commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,275 @@
+<h3 class="hd-6 example-set-hd">Warning w/o Title</h3>
+<div class="example-set">
+    <div class="alert alert-warning">
+        <div class="icon-display">
```
See [my note about simplifying this above](https://github.com/edx/ux-pattern-library/pull/277/files#r51175181).

dsjen replied:
Done.

talbs commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,275 @@
+<h3 class="hd-6 example-set-hd">Warning w/o Title</h3>
+<div class="example-set">
+    <div class="alert alert-warning">
+        <div class="icon-display">
```
See [my note about simplifying this above](https://github.com/edx/ux-pattern-library/pull/277/files#r51175181).

dsjen replied:
Done.

talbs commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,275 @@
+<h3 class="hd-6 example-set-hd">Information w/ Title</h3>
+<div class="example-set">
+    <div class="alert alert-information">
+        <div class="icon-display">
```
See [my note about simplifying this above](https://github.com/edx/ux-pattern-library/pull/277/files#r51175181).

dsjen replied:
Done.

talbs commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,275 @@
+<h3 class="hd-6 example-set-hd">Information w/o Title</h3>
+<div class="example-set">
+    <div class="alert alert-information">
+        <div class="icon-display">
```
See [my note about simplifying this above](https://github.com/edx/ux-pattern-library/pull/277/files#r51175181).

dsjen replied:
Done.

talbs commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,275 @@
+<h3 class="hd-6 example-set-hd">Information w/o Title</h3>
+<div class="example-set">
+    <div class="alert alert-information">
+        <div class="icon-display">
```
See [my note about simplifying this above](https://github.com/edx/ux-pattern-library/pull/277/files#r51175181).

dsjen replied:
Done.

talbs commented on pattern-library/sass/patterns/_alerts.scss:
```diff
@@ -0,0 +1,99 @@
+// ----------------------------
+@mixin alert-base($alert-color) {
+    border-top: rem(4) solid $alert-color;
+    .alert-icon {
```
NIT: please add an extra line here.

dsjen replied:
Added.

clrux commented:
@dsjen I can only partially review these alerts. Much of "alert" accessibility comes from how they're displayed and treated, meaning proper focus management to, within, and out of the alert, as well as how the alert relates to the content.

The markup is pretty basic here, but I would do a few things:

* add `role="alert"` to each alert
* every alert should have a title, even if it's hidden visually, and that title needs a unique `id`
* add `aria-labelledby="..."` to the alert that references the `id` of the title
* for alerts that popup in a modal/dialog, use `role="alertdialog"` instead of `role="alert"`
* alerts should be focusable, programmatically, so add `tabindex="-1"` to make this possible

Example markup:

```
<div class="alert" role="alert" aria-labelledby="alert-title" tabindex="-1">
    <h2 class="hd hd-2" id="alert-title">Halp!</h2>
    <p class="copy">And the rest, as they say, is history.</p>
</div>
```

talbs commented on pattern-library/sass/patterns/_alerts.scss:
```diff
@@ -0,0 +1,99 @@
+// #GENERAL
+// ----------------------------
+.alert {
+    border: rem(1) solid palette(grayscale-cool, light);
```
Please make this value into a theming-friendly Sass variable that can be configured and overriden. See https://github.com/edx/ux-pattern-library/blob/master/pattern-library%2Fsass%2Fpatterns%2F_lists.scss#L22 as an example of how we've done it before.

dsjen replied:
Added the config sections with variables.

talbs commented on pattern-library/sass/patterns/_alerts.scss:
```diff
@@ -0,0 +1,99 @@
+// ----------------------------
+.alert {
+    border: rem(1) solid palette(grayscale-cool, light);
+    border-radius: rem(3);
```
Please make this value into a theming-friendly Sass variable that can be configured and overriden. See https://github.com/edx/ux-pattern-library/blob/master/pattern-library%2Fsass%2Fpatterns%2F_lists.scss#L22 as an example of how we've done it before.

dsjen replied:
Added the config sections with variables.

talbs commented on pattern-library/sass/patterns/_alerts.scss:
```diff
@@ -0,0 +1,99 @@
+    padding: rem(7);
+
+    @include susy-media($bp-screen-md) {
+        float: left;
```
RTL syntax is needed here.

dsjen replied:
Do you have an example function/method for doing that?  I think I noticed one somewhere, but have no idea where it is.

talbs replied:
Sure, we documented RTL/LTR practices in our Sass wiki - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#right-to-left-rtl-support. You should find some background there as well as links to syntax and utilities. Shout if you need more help.

talbs commented on pattern-library/sass/patterns/_alerts.scss:
```diff
@@ -0,0 +1,99 @@
+    @include susy-media($bp-screen-md) {
+        float: left;
+    }
+
```
NIT: Extra line.

dsjen replied:
Thanks, updated.

talbs commented on pattern-library/sass/patterns/_alerts.scss:
```diff
@@ -0,0 +1,99 @@
+.alert-copy {
+    @extend %copy-meta;
+
+    > * {
```
I'm not sure what this selector/rule is doing. Are you removing bottom margin/padding/borders from any direct child within an alert's copy?

talbs replied:
When/why would this be needed?

dsjen replied:
Oops, removed.

talbs commented on pattern-library/sass/patterns/_alerts.scss:
```diff
@@ -0,0 +1,99 @@
+
+.alert-message {
+    padding: spacing-vertical(small);
+    padding-left: 0;
```
RTL syntax is needed here.

dsjen replied:
Thanks, updated.

talbs commented on pattern-library/sass/patterns/_alerts.scss:
```diff
@@ -0,0 +1,99 @@
+.alert-message {
+    padding: spacing-vertical(small);
+    padding-left: 0;
+    padding-right: 0;
```
RTL syntax is needed here.

dsjen replied:
Thanks, updated.

talbs commented on pattern-library/sass/patterns/_alerts.scss:
```diff
@@ -0,0 +1,99 @@
+    padding-right: 0;
+
+    @include susy-media($bp-screen-md) {
+        float: left;
```
RTL syntax is needed here.

dsjen replied:
Updated.

dsjen commented:
@clrux -- thank you!  Would a reasonable title be "Warning" or something like that or more descriptive?

clrux commented:
@dsjen The title needs to descriptively summarize the contents of the alert. The fact that it's a heading means that screenreader users can quickly navigate around the various headings if they lose place, so having something that describes the alert would be nice.

```
<div class="alert" role="alert" aria-labelledby="alert-title" tabindex="-1">
    <h2 class="hd hd-2" id="alert-title">Incorrect login</h2>
    <p class="copy">Your username and/or password did not match what we have on file.</p>
</div>
```

The above would read as:

```
Incorrect login, alert
Your username and/or password did not match.
```

You can see how the `role`, the `id` of the title, and the `aria-labelledby` work together to construct a meaningful indication of what the content is.

talbs commented on CHANGELOG.md:
```diff
@@ -5,6 +5,11 @@ Nothing yet
 
 - - -
 
+## 0.9.0 (2016-01-28)
```
:boom: Nueve!! :boom:

talbs commented on pattern-library/sass/patterns/_alerts.scss:
```diff
@@ -0,0 +1,99 @@
+
+.alert-action {
+    @include susy-media($bp-screen-md) {
+        float: right;
```
RTL syntax is needed here.

dsjen replied:
Updated.

talbs commented on pattern-library/sass/patterns/_alerts.scss:
```diff
@@ -0,0 +1,99 @@
+}
+
+.alert-action {
+    @include susy-media($bp-screen-md) {
```
NIT: please an extra line.

dsjen replied:
Done.

talbs commented on pattern-library/sass/patterns/_alerts.scss:
```diff
@@ -0,0 +1,99 @@
+// ----------------------------
+// #SETTINGS
+// ----------------------------
+@mixin alert-base($alert-color) {
```
NIT: since this is really just a mixin or an archetype we're creating alert styles from, I think the mixin name can be simplified to just be ``alert()``.

dsjen replied:
No problem -- updated!

talbs commented on pattern-library/sass/patterns/_alerts.scss:
```diff
@@ -0,0 +1,99 @@
+// ----------------------------
+
+// ----------------------------
+// #SETTINGS
```
Mind re-labelling this section and organizing this partial so that its similar to other pattern partials (https://github.com/edx/ux-pattern-library/blob/master/pattern-library%2Fsass%2Fpatterns%2F_lists.scss)? Specifically, this mixin should be under a "Utilities" area and any additional theming/configuration should be in a separate initial area called "Config".

talbs replied:
Also, we usually give a quick ToC at the top of Sass partials. Mind adding that in?

dsjen replied:
Sure thing!

talbs commented on pattern-library/sass/patterns/_alerts.scss:
```diff
@@ -0,0 +1,99 @@
+        padding-bottom: 0;
+
+        // shift the message down to be in line with the icon
+        margin-top: rem(3);
```
This seems like a magic number (https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#imperfect-and-magic-numbers). Any way you can use one of our values from the ``spacing-vertical`` Sass map instead?

dsjen replied:
Updated.

talbs commented on pattern-library/sass/patterns/_alerts.scss:
```diff
@@ -0,0 +1,99 @@
+    border-radius: 50%;
+
+    // create room around the icon for the circle
+    padding: rem(7);
```
This seems like a magic number (https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#imperfect-and-magic-numbers). Any way you can use one of our values from the ``spacing-vertical`` and ``spacing-horizontal`` Sass maps instead?

dsjen replied:
Updated.

talbs commented on pattern-library/sass/patterns/_alerts.scss:
```diff
@@ -0,0 +1,99 @@
+}
+
+.alert-icon {
+    // create a circle around the icon
```
Sweet comments and inline doc!

talbs commented on pattern-library/sass/patterns/_alerts.scss:
```diff
@@ -0,0 +1,99 @@
+
+// information-based alert
+.alert-information {
+    @include alert-base(palette(information, base));
```
Ideally your references to the ``alert()`` mixin should also use Sass variables/configuration rather than direct colors. Once you set up some alert-specific Sass variables, including alert type colors, mind changing these rules to be something more like the following?

```scss
.alert-information {
     @include alert($alert-information-color);
}
```

talbs commented on pldoc/static/sass/_components.scss:
```diff
@@ -44,69 +44,6 @@
 // ------------------------------
 // #SYSTEM FEEDBACK
 // ------------------------------
-%message {
```
Great clean up and refactor!

talbs commented:
@dsjen, took a spin through your front end (templates, posts, html, and Sass/CSS). Nice work getting things up and running as your first pattern. I left some inline feedback, let me know if anything is confusing or you have concerns. I'll be poking through the visual design and rendering now.

talbs commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,275 @@
+<div class="example-set">
+    <div class="alert alert-error">
+        <div class="icon-display">
+            <span class="icon alert-icon icon-check" aria-hidden="true"></span>
```
NIT: incorrect icon for the occasion (error).

dsjen replied:
Thanks!  Updated.

talbs commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,275 @@
+<div class="example-set">
+    <div class="alert alert-success">
+        <div class="icon-display">
+            <span class="icon alert-icon icon-exclamation-triangle" aria-hidden="true"></span>
```
NIT: incorrect icon for the occasion (success).

dsjen replied:
Thanks, updated!

dsjen commented:
I believed I've addressed all comments.  Please let me know if there's anything else.

clrux commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,108 @@
+            </h3>
+        </div>
+
+        <button type="button" class="alert-action btn-neutral btn">Read More</button>
```
Please make this button text more descriptive so it makes sense if taken out of context. In this case "Read more" is dependent upon surrounding content, but screen readers allow users to pull up a list of all form controls, so seeing a bunch of "Read more"'s won't be very helpful.

clytwynec replied:
@clrux Might this be an appropriate place to use '.sr' text?  I could make a case for not wanting to visually add more text than needed to the button.

clrux replied:
Yep, great idea. So, @dsjen you could do something like:

```
<button>Read more <span class="sr">about charts</button
```

Or the like.

dsjen replied:
Make sense!

clrux commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,108 @@
+
+        <div class="alert-message">
+            <h3 class="alert-title" id="alert-title-success">
+                Success
```
Since this title is what' announced first, I suggest giving it more context. Something like "Your changes have been saved" would work. You could then use the copy below to add any relevant information that wouldn't be necessary to understanding the alert. Basically, if you can say what you need to say in a few works, let that be the title.

dsjen replied:
Thanks -- updated to new copy.

clrux commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,108 @@
+            </p>
+        </div>
+
+        <button type="button" class="alert-action btn-neutral btn">Reload</button>
```
Provide more context for this button's actions.

clrux replied:
NIT: double spacing after periods

clrux commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,108 @@
+<div class="example-set">
+    <div class="alert alert-warning" role="alert" aria-labelledby="alert-warning-message" tabindex="-1">
+        <span class="icon alert-icon icon-exclamation-triangle" aria-hidden="true"></span>
+
```
This needs a heading title with an `.sr` class. Have the `aria-labelledby` reference it instead of the `p`.

dsjen replied:
@clrux, are you recommending that a title be included with the `sr` class and that it not be visible?

clrux replied:
@dsjen Every alert needs a title. Whether it's visible or not is up to you.

talbs commented:
@dsjen, thanks. Here's some persnickety UI and visual design feedback:

**Small Visual Discrepancies**
![image](https://cloud.githubusercontent.com/assets/163763/12680896/134c95d8-c67a-11e5-8940-35458f230949.png)

From a quick screengrab of a portion the static designs, there a few things to tidy up:

* the top border width should be the equivalent of ``2px`` (so ``rem(2)`` will get you there)
* the top left and right border-radius should be 0 (see https://developer.mozilla.org/en-US/docs/Web/CSS/border-top-left-radius for how to target those)
* there's a slight box shadowon all alerts in the original comps (if it helps, the value is ``0 2 2 0``  and the color used is ``palette(grayscale-cool, xx-light)`` in Sass/CSS)

**RE: Repsonsive/smaller breakpoint layouts**
We agreed in person to try hiding the icon to help make the alert not take up as much visual space. Mind getting that in?

**Content/Examples**
* Can you provide an example that has a link within its message/copy?
* Can you provide an example that has no icon (for the group to review if that case is needed)?
* It might be worthwhile to label these current examples as "View-level Alerts" to note that they should be used for larger areas of content/transactions within a view. We'll have other alerts coming in the future (e.g. inline messages and banner/app-wide alerts)

Let me know if you need any more context or want to chat anything out.

talbs commented on pattern-library/sass/patterns/_alerts.scss:
```diff
@@ -0,0 +1,115 @@
+// #GENERAL
+// ----------------------------
+.alert {
+    border: $alert-border;
```
@dsjen, one thing I'm not seeing here is a ``background-color`` property. You may want to add that (and the appropriate Sass configuration variable, for theming purposes). I think the Doc Site's rendering might be giving you that white background for free in our previews, but that needs to be provided for real use. Make sense?

dsjen commented:
@talbs -- I've updated the styles, created and example without icon, hidden the icons on small screens, etc.

@clytwynec & @clrux -- I've added the `sr-only` class (rather than `sr`).

chris-mike commented:
I wrote introductory copy for alerts, which is on the wiki page: https://openedx.atlassian.net/wiki/display/FEDX/Alert+Pattern

dsjen commented:
Thanks, @chris-mike!  I've pushed the update.

![screen shot 2016-02-02 at 2 59 37 pm](https://cloud.githubusercontent.com/assets/5265058/12762706/7b38c2b0-c9be-11e5-83ed-b4ceee5a2d4c.png)

andy-armstrong commented on _posts/patterns/2016-01-28-alerts.md:
```diff
@@ -0,0 +1,112 @@
+</div>
+
+
+<h3 class="hd-6 example-set-hd">Warning Alert with Message Only and No Icon</h3>
```
I'm not sure about this use case. Is the intention that this is an alert that the user shouldn't pay much attention to? It seems too subtle to me, in that it is only the very thin yellow top border that distinguishes this from a non-alert. I would be tempted to add the warning icon back, but leave it without a title so that the text doesn't look like a heading. In fact, I then wonder whether the first example should use a message and not a title.

dsjen replied:
It does look rather plain with the icon, and I'm happy to remove that example entirely.

Regarding the first example, I think there's utility in including an example with a title only.  Is it the content that you're reacting to, the usage, or something else entirely?

andy-armstrong replied:
Thanks, I think just adding an icon will be great.

For the first example, I think I am reacting to the content. The title seems more like a message, and I find it slightly jarring to read messages in a bold font. It is only a very minor point so if folks are fine with it then that's okay.

andy-armstrong commented on pattern-library/sass/patterns/_alerts.scss:
```diff
@@ -0,0 +1,132 @@
+// #CONFIG
+// ----------------------------
+$alert-background-color: palette(grayscale, white) !default;
+$alert-border-radius: rem(3) !default;
```
Should we be using a variable rather than a hard-coded value of 3? Maybe it should use one of the spacing variables:

https://github.com/edx/ux-pattern-library/blob/master/pattern-library/sass/global/_settings.scss#L305

dsjen replied:
Thank you!  I'll update this to use a variable.

andy-armstrong commented on pattern-library/sass/patterns/_alerts.scss:
```diff
@@ -0,0 +1,132 @@
+    border-top:  rem(2) solid $alert-color;
+
+    .alert-icon {
+        color: palette(grayscale, white);
```
Should the white be themable, i.e. a variable?

dsjen replied:
Good point!  I'll add `$alert-icon-color` to the config.

andy-armstrong commented on pattern-library/sass/patterns/_alerts.scss:
```diff
@@ -0,0 +1,132 @@
+@mixin alert-message($width) {
+
+    @include susy-media($bp-screen-md) {
+        @include float(left);
```
Is the float function left-to-right aware? Do we need to do something to make this work right-to-left?

dsjen replied:
Yes, this uses [Bi App Sass](http://anasnakawa.github.io/bi-app-sass/), and I just added documentation to our [wiki about RTL support](https://openedx.atlassian.net/wiki/display/FEDX/How+to+add+a+pattern+to+the+UX+Pattern+Library#HowtoaddapatterntotheUXPatternLibrary-Right-to-Left(RTL)).

andy-armstrong commented on pattern-library/sass/patterns/_alerts.scss:
```diff
@@ -0,0 +1,132 @@
+    width: 100%;
+
+    @include susy-media($bp-screen-md) {
+        @include float(right);
```
Does this also need rtl handling?

dsjen replied:
Please see previous comment re: rtl.

andy-armstrong commented on pattern-library/sass/patterns/_alerts.scss:
```diff
@@ -0,0 +1,132 @@
+    display: none;
+
+    @include susy-media($bp-screen-md) {
+        @include float(left);
```
Does this also need rtl handling?

dsjen replied:
Please see previous comment re: rtl.

andy-armstrong commented:
@dsjen This looks gorgeous! I'm finished with my first review, so let me know when I should look again.

andy-armstrong commented:
FYI @AlasdairSwan @aleffert

dsjen commented:
@andy-armstrong -- thank you!  I've added SASS variables to the config, and added the icon back in for the warning example.  Please let me know if there's anything else.

@clrux -- have I addressed your concerns?

andy-armstrong commented:
@dsjen Last nit: can you get the icon and the message to align when there is no title? As it stands, there seems to be too much vertical space above the message:

![image](https://cloud.githubusercontent.com/assets/5985072/12765644/9ac62302-c9cd-11e5-9b84-e5d14970f6ef.png)

andy-armstrong commented:
@dsjen In fact, there is too much padding both above and below the message. It would be great to tighten this up.

andy-armstrong commented:
:+1: with that change.

dsjen commented:
@andy-armstrong --  concerning the alignment of the message only example to the icon, I'm not sure I see the alignment problem.  I was trying to center them.  Do you think it should be bottom aligned rather?

In regards to the spacing, I ended up tightening up the horizontal spacing a tad (see Small Spacing) and tried an x-small spacing, but it looked a bit too tight to me (see X-Small Spacing).  There is a bit more room at the bottom because the message is wrapped in a paragraph tag.  I could modify the spacing for alert paragraphs, but that might be for multi-paragraph alerts (will that even happen?), the paragraphs wouldn't have much separation.

## Small Spacing
![sm](https://cloud.githubusercontent.com/assets/5265058/12766094/e1a059c6-c9cf-11e5-8def-8ee386f52a1b.png)

## X-Small Spacing
![x-sm](https://cloud.githubusercontent.com/assets/5265058/12766095/e1a6060a-c9cf-11e5-9c0d-d48e3df84e99.png)

andy-armstrong commented:
Just for the record, @dsjen and I spoke about the alignment. He's going to look into matching up with Brian's design where the titles were not vertically centered, but were instead top aligned with the icon.

![image](https://cloud.githubusercontent.com/assets/5985072/12767517/989547d4-c9d7-11e5-8252-b1bf2252b309.png)

For messages we'll leave them as they are. It looks better for a single line message to be centered vertically, but when it wraps it would look better top aligned. We don't believe there's a CSS way to adjust the margin based upon whether an item wraps, but Dennis will investigate and use it if possible.

clrux commented:
I would check contrast here, especially the lighter gray text.

dsjen commented:
I've pushed (what I believe is) the final update.

@clrux -- Thanks for checking on the color contrast!  I ran aXe on the page and it didn't call out any errors with the alerts component.  If you've completed the review, would you please check the "accessibility"  box at the top of the PR?  I'd be delighted to wrap this up today.

clrux commented:
Waiting for peer responses (warning), reload button says "Reload about assessments"

Otherwise, nice job. Once you fix that button, :+1:

dsjen commented:
@clrux -- good catch!  It's updated now.
